umm-spec-lib/src/cmr/umm_spec/location_keywords.clj

```diff
@@ -43,20 +43,15 @@
 (defn- location-values
   "Returns the location keyword values in order so that we can get the last one"
   [location-keyword]
-  (for [k location-keyword-order
-        :let [value (get location-keyword k)]
-        :when value]
-    value))
-
-(defn- leaf-value
-  "Returns the leaf value of the location-keyword object to be put in a SpatialKeywords list"
-  [location-keyword]
-  (last (location-values location-keyword)))
+  (as-> location-keyword keyword
+        (vals keyword)
+        (distinct keyword)
+        (remove nil? keyword)))
 
 (defn location-keywords->spatial-keywords
   "Converts a list of LocationKeyword maps to a list of SpatialKeywords"
   [location-keyword-list]
-  (map #(leaf-value %) location-keyword-list))
+  (flatten (map location-values location-keyword-list)))
```

Comment:
I looked some more - I don't see a problem with indexing all of the fields from the location keyword hierarchy so that a spatial keyword search will return a match of any string in the hierarchy.

By the way in this case I think `mapcat` is the idiomatic way rather than `(flatten (map`.